src/DeviceIdWidget/widget/DeviceIdWidget.xml

```diff
@@ -0,0 +1,68 @@
+    <properties>
+        <property key="deviceEntity" type="entity" required="true">
+            <caption>Device entity</caption>
+            <category>Device</category>
+            <description></description>
+        </property>
+        <property key="deviceIDAttribute" type="attribute" entityProperty="deviceEntity" required="false">
```

Comment:
I do agree on camel casing of acronyms, use `deviceIdAttribute` instead of `deviceIDAttribute`